### PR TITLE
Add Checkbox to simplify Smalltalk CI Logs

### LINKED ISSUE #89: Checkbox to Toggle SmalltalkCI Parsing
As a <kbd>*user*</kbd> I want to *be able to deactivate SmalltalkCI parsing in the ResultsWindow* in order to *see the raw format of the displayed text in case the parsing is faulty*.

### Acceptance criteria

- [x] In the ResultsWindow, there is a checkbox labeled "Parse SmalltalkCI"
- [x] When checking the box, SmalltalkCI parsing is activated
- [x] When unchecking the box, SmalltalkCI parsing is deactivated

### Effort Evaluation

Points: 1

### PR DESCRIPTION
**Pull request by benjamin-frost**
Closes: #89

### CODE REVIEW

File: packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/buildSimplifySmalltalkCiLogCheckBoxWith..st [hunk 1]
```diff
@@ -0,0 +1,9 @@
+building
+buildSimplifySmalltalkCiLogCheckBoxWith: aBuilder
+
+	^ aBuilder pluggableCheckBoxSpec new
+		model: self;
+		label: #simplifySmalltalkCiLogCheckBoxLabel;
+		action: #simplifySmalltalkCiLogCheckBoxAction;
+		state: #simplifySmalltalkCiLogCheckBoxState;
+		yourself
```
Comment (SebastianWagner2) [inline packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/buildSimplifySmalltalkCiLogCheckBoxWith..st]:
```suggestion
building
buildSimplifyLogCheckBoxWith: aBuilder

	^ aBuilder pluggableCheckBoxSpec new
		model: self;
		label: #simplifyLogCheckBoxLabel;
		action: #simplifyLogCheckBoxAction;
		state: #simplifyLogCheckBoxState;
		yourself
```
I felt the names were a little too long to support readability. Although the parsing is in the same class i would not include any info on implementation (SmalltalkCiLog vs Log) in GUI code

File: packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/shouldSimplifySmalltalkCiLog..st [hunk 2]
```diff
@@ -0,0 +1,4 @@
+accessing
+shouldSimplifySmalltalkCiLog: aBoolean
+	
+	shouldSimplifySmalltalkCiLog := aBoolean
```
Comment (SebastianWagner2) [inline packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/shouldSimplifySmalltalkCiLog..st]:
```suggestion
accessing
shouldSimplifyLog: aBoolean
	
	shouldSimplifyLog := aBoolean
```
I felt the names were a little too long to support readability. Although the parsing is in the same class i would not include any info on implementation (SmalltalkCiLog vs Log) in GUI code

File: packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/shouldSimplifySmalltalkCiLog.st [hunk 3]
```diff
@@ -0,0 +1,4 @@
+accessing
+shouldSimplifySmalltalkCiLog
+	
+	^ shouldSimplifySmalltalkCiLog ifNil: [ shouldSimplifySmalltalkCiLog := false ]
```
Comment (SebastianWagner2) [inline packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/shouldSimplifySmalltalkCiLog.st]:
```suggestion
accessing
shouldSimplifyLog
	
	^ shouldSimplifyLog ifNil: [ shouldSimplifyLog := false ]
```
I felt the names were a little too long to support readability. Although the parsing is in the same class i would not include any info on implementation (SmalltalkCiLog vs Log) in GUI code

File: packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/simplifySmalltalkCiLogCheckBoxAction.st [hunk 4]
```diff
@@ -0,0 +1,5 @@
+building
+simplifySmalltalkCiLogCheckBoxAction
+	
+	self shouldSimplifySmalltalkCiLog: self shouldSimplifySmalltalkCiLog not.
+	self updateJobLogForId: ( (self jobs at: selectedJob) id )
```
Comment (SebastianWagner2) [inline packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/simplifySmalltalkCiLogCheckBoxAction.st]:
```suggestion
some other category not sure
simplifyLogCheckBoxAction
	
	self shouldSimplifyLog: self shouldSimplifyLog not.
	self updateJobLogForId: (self jobs at: selectedJob) id
```
I felt the names were a little too long to support readability. Although the parsing is in the same class i would not include any info on implementation (SmalltalkCiLog vs Log) in GUI code

Idk if the parenthesis are necessary

File: packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/simplifySmalltalkCiLogCheckBoxFrame.st [hunk 5]
```diff
@@ -0,0 +1,9 @@
+building
+simplifySmalltalkCiLogCheckBoxFrame
+	
+	^ LayoutFrame new
+		leftFraction: 0.3 offset: 0;
+		topFraction: 0.3 offset: 0;
+		rightFraction: 1 offset: 0;
+		bottomFraction: 0.3 offset: self class defaultCheckBoxHeight;
+		yourself
```
Comment (SebastianWagner2) [inline packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/simplifySmalltalkCiLogCheckBoxFrame.st]:
```suggestion
building
simplifyLogCheckBoxFrame
	
	^ LayoutFrame new
		leftFraction: 0.3 offset: 0;
		topFraction: 0.3 offset: 0;
		rightFraction: 1 offset: 0;
		bottomFraction: 0.3 offset: self class defaultCheckBoxHeight;
		yourself
```
I felt the names were a little too long to support readability. Although the parsing is in the same class i would not include any info on implementation (SmalltalkCiLog vs Log) in GUI code

File: packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/simplifySmalltalkCiLogCheckBoxLabel.st [hunk 6]
```diff
@@ -0,0 +1,4 @@
+building
+simplifySmalltalkCiLogCheckBoxLabel
+	
+	^ 'Simplify SmalltalkCI Log'
```
Comment (SebastianWagner2) [inline packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/simplifySmalltalkCiLogCheckBoxLabel.st]:
```suggestion
accessing
simplifyLogCheckBoxLabel
	
	^ 'Simplify ' , _fetch String here instead_ , ' Log'
```
I felt the names were a little too long to support readability. Although the parsing is in the same class i would not include any info on implementation (SmalltalkCiLog vs Log) in GUI code

Would put this String in this class to allow for reusability for potential simplification option of travis status log (then we could move all this code to ATDDRemoteResults)

File: packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/simplifySmalltalkCiLogCheckBoxState.st [hunk 7]
```diff
@@ -0,0 +1,4 @@
+building
+simplifySmalltalkCiLogCheckBoxState
+	
+	^ self shouldSimplifySmalltalkCiLog
```
Comment (SebastianWagner2) [inline packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/simplifySmalltalkCiLogCheckBoxState.st]:
```suggestion
accessing
simplifyLogCheckBoxState
	
	^ self shouldSimplifyLog
```
I felt the names were a little too long to support readability. Although the parsing is in the same class i would not include any info on implementation (SmalltalkCiLog vs Log) in GUI code

File: packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/updateJobLogForId..st [hunk 8]
```diff
@@ -7,8 +7,11 @@ updateJobLogForId: aNumber
 		on: GHAAccessDeniedError 
 		do: [ :error |  response := 'Acces denied. Logs are only available when providing a token' ].
 		
-	(self isSmalltalkCILog: response)
-		ifTrue: [response := self parseSmalltalkCILog: response].
+	(self shouldSimplifySmalltalkCiLog)
+		ifTrue: [
+			(self isSmalltalkCILog: response)
+				ifTrue: [response := self parseSmalltalkCILog: response].
+		].
```
Comment (SebastianWagner2) [inline packages/AutoTDD-GUI.package/ATDDGitHubActionResults.class/instance/updateJobLogForId..st]:
```suggestion
	(self shouldSimplifyLog)
		ifTrue: [
			(self isSmalltalkCILog: response)
				ifTrue: [response := self parseSmalltalkCILog: response].
		].
```
I felt the names were a little too long to support readability. Although the parsing is in the same class i would not include any info on implementation (SmalltalkCiLog vs Log) in GUI code

File: packages/AutoTDD-GUI.package/ATDDTravisStatusResults.class/instance/buildJobLogComponentsWith..st [hunk 9]
```diff
@@ -0,0 +1,8 @@
+building
+buildJobLogComponentsWith: aBuilder
+	
+	^ OrderedCollection new
+		add: ((self buildJobLogTextWith: aBuilder)
+				frame: self jobLogTextFrame;
+				yourself);
```
Comment (SebastianWagner2) [inline packages/AutoTDD-GUI.package/ATDDTravisStatusResults.class/instance/buildJobLogComponentsWith..st:7]:
```suggestion
			frame: self jobLogTextFrame;
			yourself);
```

**Review by SebastianWagner2 — CHANGES_REQUESTED** [review]

**Review by kolioOtSofia — APPROVED** [review]

**Review by SebastianWagner2 — APPROVED** [review]

**Pull request merged**